### PR TITLE
[MPS] Add test consistency from OpInfo based tests from PR 78504

### PR DESCRIPTION
**Pull request by kulinseth**

### CODE REVIEW

**Review by malfet — COMMENTED** [review]
This looks pretty identical to This looks very much identical to https://github.com/pytorch/pytorch/pull/79521
Also, change can not be landed as is, as `pprint` is not a standard dependnecy

And it would be nice if it mentioned issue it fixes

**Comment by kulinseth** [comment]
> This looks pretty identical to This looks very much identical to #79521 Also, change can not be landed as is, as `pprint` is not a standard dependnecy
> 

Unfortunately it turned out to be the same underlying issue. The issue was related to correctness which we noticed in constan_pad_nd operation. I have approved #79521, after it merges, I will rebase to check-in the test_consistency tests.

**Comment by kulinseth** [comment]
Rebased the changes, now the PR has only the test consistency in test_mps. @albanD , please take a look.

File: test/test_mps.py
Comment (albanD) [inline test/test_mps.py:4425]: I guess this is copy pasted? But I think this if statement should be only true in this context.

File: test/test_mps.py
Comment (albanD) [inline test/test_mps.py:4449]: I this case don't you have `t is v._base` ? 
(same for all the tests below)

File: test/test_mps.py
Comment (albanD) [inline test/test_mps.py:4710]: is the is view check?

File: test/test_mps.py
Comment (albanD) [inline test/test_mps.py:4414]: Can you please add a big comment here that this bloc should be removed once the corresponding device generic test can be made to run on MPS. It would be good to link exactly which part needs to be updated.
And if anything here was added beyond the original test, I think we want to break it into another class.

File: test/test_mps.py
Comment (albanD) [inline test/test_mps.py:5040]: Does that still not work?

File: test/test_mps.py
Comment (albanD) [inline test/test_mps.py:5194]: Why is this being removed?

File: test/test_mps.py
Comment (albanD) [inline test/test_mps.py:5733]: I think this groups is the same as the one below, they can be merged.

File: test/test_mps.py
Comment (kulinseth) [reply test/test_mps.py:4710]: Yes.

File: test/test_mps.py
Comment (kulinseth) [reply test/test_mps.py:4414]: These are all subset of test/test_view_ops.py tests.

File: test/test_mps.py
Comment (kulinseth) [reply test/test_mps.py:5194]: Added it back.

File: test/test_mps.py
Comment (kulinseth) [reply test/test_mps.py:5040]: Its working.

File: test/test_mps.py
Comment (kulinseth) [reply test/test_mps.py:5733]: Right, I want to separate it out from the group which is `flaky`. These test are consistently failing. I can remove the comment. They are same as the tests above.

**Review by albanD — APPROVED** [review]
Thanks for the update!

**Comment by kulinseth** [comment]
```
File "/Users/gharunner/actions-runner/_work/pytorch/pytorch/conda-test-env-2519578557/lib/python3.8/site-packages/torchgen/utils.py", line [34](https://github.com/pytorch/pytorch/runs/6946646878?check_suite_focus=true#step:6:35), in <module>
34
    from yaml import CSafeLoader as Loader
[35](https://github.com/pytorch/pytorch/runs/6946646878?check_suite_focus=true#step:6:36)
ModuleNotFoundError: No module named 'yaml'
[36](https://github.com/pytorch/pytorch/runs/6946646878?check_suite_focus=true#step:6:37)
```

How do we enable yaml dependency for MPS. @albanD

**Comment by frank-wei** [comment]
The third party libs are overridden. It is not intentional I think. Please revert it.
cc @kulinseth @malfet @albanD

**Review by malfet — APPROVED** [review]
LGTM, but please undo submodule updates (which aren't needed, are they?)

**Pull request closed without merging**
